Proteomics/Protein/Protein.cs

```diff
@@ -227,33 +228,38 @@ public string GetEnsemblFastaHeader()
         /// </summary>
         public IEnumerable<PeptideWithSetModifications> Digest(DigestionParams digestionParams, IEnumerable<Modification> allKnownFixedModifications,
             List<Modification> variableModifications, List<SilacLabel> silacLabels = null, (SilacLabel startLabel, SilacLabel endLabel)? turnoverLabels = null)
-        {
+        {            
             //can't be null
             allKnownFixedModifications = allKnownFixedModifications ?? new List<Modification>();
+            List<Modification> allFixedModsList = allKnownFixedModifications.ToList();
```

Comment:
creating a new list for each time `Digest` is called is going to create memory/garbage collection problems. I would just change `IEnumerable<Modification> allKnownFixedModifications` in the parameters to `List<Modification> allKnownFixedModifications`. then check `if (digestionParams.Protease.CleavageMod != null && allKnownFixedModifications.Contains(digestionParams.Protease.CleavageMod))`